Add support for multiple trackers

MrBr opened the pull request.

MrBr committed:
Reexport native bridges

MrBr committed:
Improve GoogleAnalytics service

- Add support for multiple trackers
- Move class to separate file

MrBr committed:
Create GoogleAnalyticsSettings class

- Used for shared GA settings

MrBr committed:
Move GoogleTagManager to separate file

MrBr committed:
Reexport GoogleAnalytics services

MrBr committed:
Simplify GA bridge method use

Vunovati commented on src/GoogleAnalytics.js:
```diff
@@ -0,0 +1,252 @@
+   * @param {String} trackerId
+   */
+  addTrackerId(trackerId) {
+    this.trackerIds.push(trackerId);
```
The Set does not have a push method. Are you sure this would work?

MrBr replied:
It will not work, this is error, worked with arrays before so this push left from old code. Fixed.

MrBr committed:
Improve GoogleAnalytics service

- Fix addTrackerId
- Rename callNativeMethod to callNativeGoogleanalyticsMethod

MrBr committed:
Improvements

MrBr committed:
Create GoogleAnalyticsTracker class

MrBr committed:
Create GoogleAnalyticsBackwardsCompatibility class

MrBr committed:
Change backwards compatibility

- Remove proxy and implement needed methods

MrBr committed:
Improve Tracker customDimensions handling

MrBr committed:
Improve GA settings comment

MrBr committed:
Create analytics setSamplingRate native function

MrBr committed:
Add customDimensions type check

Vunovati commented on src/GoogleAnalyticsTracker.js:
```diff
@@ -1,103 +1,53 @@
+  transformCustomDimensionsFieldsToIndexes(customDimensions) {
+    if (this.customDimensionsFieldsIndexMap) {
+      const mappedCustomDimensions = {};
+      Object.keys(this.customDimensionsFieldsIndexMap).forEach(key => {
```
it would be nicer if you used filter and reduce here.

MrBr replied:
Done.

Vunovati commented on src/GoogleAnalyticsTracker.js:
```diff
@@ -1,103 +1,53 @@
    * Track the current screen/view
    * @param  {String} screenName The name of the current screen
    */
   trackScreenView(screenName) {
```
do you need screenName here if you are forwarding all arguments?

MrBr replied:
I want to keep signature visible, that is why I kept arguments, bot sure if IDEs will suggest it only based on JSDoc.

MrBr committed:
Improve transformCustomDimensionsFieldsToIndexes
